code/model/modelread.cpp

```diff
@@ -2290,7 +2306,41 @@ int read_model_file(polymodel * pm, char *filename, int n_subsystems, model_subs
 		id = cfread_int(fp);
 		len = cfread_int(fp);
 		next_chunk = cftell(fp) + len;
+	}
 
+	// Now that we've processed all the chunks, resolve the look_at submodels if we have any
+	if (!look_at_submodel_names.empty()) {
+		for (i = 0; i < pm->n_models; i++) {
+			if (pm->submodel[i].look_at_submodel >= 0) {
+				const char *submodel_name = look_at_submodel_names[pm->submodel[i].look_at_submodel].c_str();
+
+				// search for this submodel name among all submodels
+				for (j = 0; j < pm->n_models; j++) {
+					if (!strcmp(submodel_name, pm->submodel[j].name)) {
+						nprintf(("Model", "NOTE: Matched %s %s $look_at: target %s with subobject id %d\n", pm->filename, pm->submodel[i].name, submodel_name, j));
+
+						// set the correct submodel reference, and set the char* to null as a found-flag
+						pm->submodel[i].look_at_submodel = j;
+						submodel_name = nullptr;
+						break;
+					}
+				}
+
+				// did we fail to find it?
+				if (submodel_name != nullptr) {
+					Warning(LOCATION, "Unable to match %s %s $look_at: target %s with submodel!\n", pm->filename, pm->submodel[i].name, submodel_name);
+					pm->submodel[i].look_at_submodel = -1;
+				}
+				// are we navel-gazing?
+				else if (pm->submodel[i].look_at_submodel == i) {
+					Warning(LOCATION, "Match %s %s $look_at: target %s with its own submodel!  Submodel cannot look at itself!\n", pm->filename, pm->submodel[i].name, submodel_name);
+					pm->submodel[i].look_at_submodel = -1;
+				}
+			}
+		}
+
+		// done with temporary name vector
+		look_at_submodel_names.clear();
```

Comment:
Clearing the vector isn't necessary because it will be deallocated anyway when the function finishes.